File: src/main/java/Spring/Entities/Musicians.java
package Spring.Entities;

import javax.persistence.*;
import java.util.Date;


@Entity
@Table(name = "Musicans", uniqueConstraints = {
        @UniqueConstraint(columnNames = {"id"})
})
public class Musicians {



    @GeneratedValue
    @Id
    @Column(name = "id")
    private int id;

    @Column(name = "First_Name")
    private String firstName;

    @Column(name = "Last_name")
    private String lastName;

    @Column(name = "Pseudonym")
    private String pseudonym;

    @OneToOne(cascade = CascadeType.PERSIST)
    @JoinColumn(name = "Band", referencedColumnName = "id")
    private Bands band;

    @Column(name = "Birth_date")
    private Date birthDate;

    @Column(name = "Origin")
    private String origin;

    @Column(name = "Instrument")
    private String instrument;

    @Column(name = "Average_Rate_Of_Album")
    private float averageRateOfAlbum;

    public Musicians() {
    }

    public Musicians(String firstName, String lastName, String pseudonym, Bands band, Date birthDate, String origin, String instrument, float averageRateOfAlbum) {
        this.firstName = firstName;
        this.lastName = lastName;
        this.pseudonym = pseudonym;
        this.band = band;
        this.birthDate = birthDate;
        this.origin = origin;
        this.instrument = instrument;
        this.averageRateOfAlbum = averageRateOfAlbum;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getFirstName() {
        return firstName;
    }

    public void setFirstName(String firstName) {
        this.firstName = firstName;
    }

    public String getLastName() {
        return lastName;
    }

    public void setLastName(String lastName) {
        this.lastName = lastName;
    }

    public String getPseudonym() {
        return pseudonym;
    }

    public void setPseudonym(String pseudonym) {
        this.pseudonym = pseudonym;
    }

    public Bands getBand() {
        return band;
    }

    public void setBand(Bands band) {
        this.band = band;
    }

    public Date getBirthDate() {
        return birthDate;
    }

    public void setBirthDate(Date birthDate) {
        this.birthDate = birthDate;
    }

    public String getOrigin() {
        return origin;
    }

    public void setOrigin(String origin) {
        this.origin = origin;
    }

    public String getInstrument() {
        return instrument;
    }

    public void setInstrument(String instrument) {
        this.instrument = instrument;
    }

    public float getAverageRateOfAlbum() {
        return averageRateOfAlbum;
    }

    public void setAverageRateOfAlbum(float averageRateOfAlbum) {
        this.averageRateOfAlbum = averageRateOfAlbum;
    }


}

File: FrontEnd/src/app/Class/musicians.ts
import { Bands } from './bands'

export class Musicians{

	private id: number;
	private firstName: String;
	private lastName: String;
	private pseudonym: String;
	private band: String;
	private birthDate: Date;
	private origin: String;
	private instrument: String;
	private averageRateOfAlbum: number;

	constructor(obj?: any){
	
		this.id = (obj && obj.id) || 0;
		this.firstName = (obj && obj.firstName) || '';
		this.lastName = (obj && obj.lastName) || '';
		this.pseudonym = (obj && obj.pseudonym) || '';
		this.band = (obj && obj.band) || '';
		this.birthDate = (obj && obj.birthDate) || new Date;
		this.origin = (obj && obj.origin) || '';
		this.instrument = (obj && obj.instrument) || '';
		this.averageRateOfAlbum = (obj && obj.averageRateOfAlbum) || 0;

	}

	getId(): number{
		return this.id;
	}
	setId(id: number){
		this.id = id;
	}
	
	getFirstName(): String{
		return this.firstName;
	}
	setFirstName(firstName: String){
		this.firstName = firstName;
	}

	getLastName(): String{
		return this.lastName;
	}
	setLastName(lastName: String){
		this.lastName = lastName;
	}

	getPseudonym(): String{
		return this.pseudonym;
	}
	setPseudonym(pseudonym: String){
		this.pseudonym = pseudonym;
	}
	
	getBand(): String{
		return this.band;
	}
	setBand(band: String){
		this.band = band;
	}
	
	getBirthDate(): Date{
		return this.birthDate;
	}
	setBirthDate(birthDate: Date){
		this.birthDate = birthDate;
	}

	getOrigin(): String{
		return this.origin;
	}
	setOrigin(origin: String){
		this.origin = origin;
	}

	getInstrument(): String{
		return this.instrument;
	}
	setInstrument(instrument: String){
		this.instrument = instrument;
	}

	getAverageRateOfAlbum(): number{
		return this.averageRateOfAlbum;
	}
	setAverageRateOfAlbum(averageRateOfAlbum: number){
		this.averageRateOfAlbum = averageRateOfAlbum;
	}



}